crates/kornia-imgproc/src/draw.rs

```diff
@@ -70,9 +70,114 @@ pub fn draw_line<const C: usize>(
+    color: [u8; C],
+    thickness: usize,
+) {
+    if points.len() < 3 {
+        return;
+    }
+    let num_points = points.len();
+    for i in 0..num_points {
+        let start = points[i];
+        // mod to close the polygon
+        let end = points[(i + 1) % num_points];
+        draw_line(img, start, end, color, thickness);
+    }
+}
+
+/// Draws a colored polygon on an image inplace.
+/// Uses the scanline-fill algorithm.
+///
+/// # Arguments
+///
+/// * `img` - The image to draw on.
+/// * `points` - A slice of points representing the vertices of the polygon inn order.
+/// * `color` - The color filled in the polygon drawn.
+/// * `line_color` - The color of the pen used to draw the polygon.
+/// * `thickness` - The thickness of the polygon lines.
+///
+pub fn draw_filled_polygon<const C: usize>(
+    img: &mut Image<u8, C>,
+    points: &[(i64, i64)],
+    fill_color: [u8; C],
+    line_color: [u8; C],
+    thickness: usize,
+) {
+    let n = points.len();
+    if n < 3 {
+        return;
+    }
+
+    // adding edges to edge list.
+    let mut edges = Vec::with_capacity(n);
+    let mut max_y = i64::MIN;
+    let mut min_y = i64::MAX;
+    for i in 0..n {
+        let (x0, y0) = points[i];
+        let (x1, y1) = points[(i + 1) % n];
+        edges.push((x0, y0, x1, y1));
+
+        // bounds for y to find range of scanlines
+        if y0 < min_y {
+            min_y = y0;
+        }
+        if y0 > max_y {
+            max_y = y0;
+        }
+    }
+
+    let mut x_intersections = Vec::with_capacity(n);
+
+    // for each scanline
+    for y in min_y..=max_y {
+        x_intersections.clear();
+        for &(x0, y0, x1, y1) in &edges {
+            // checking if the scanline intersects the edges.
+            if (y0 <= y && y1 > y) || (y1 <= y && y0 > y) {
+                // finding the intersection with linear interpolation
+                let x = x0 as f64
+                    + (y as f64 - y0 as f64) * (x1 as f64 - x0 as f64) / (y1 as f64 - y0 as f64);
+                x_intersections.push(x);
+            }
+        }
+
+        // sort the intersections by x coordinate
+        x_intersections.sort_by(|a, b| a.partial_cmp(b).unwrap());
+
+        // filling the region between intersections with color
+        for pair in x_intersections.chunks(2) {
+            if let [x_start, x_end] = pair {
+                let xs = x_start.ceil() as i64;
+                let xe = x_end.floor() as i64;
+                for x in xs..=xe {
+                    if x >= 0 && x < img.cols() as i64 && y >= 0 && y < img.rows() as i64 {
+                        let base = (y * img.cols() as i64 + x) * C as i64;
+                        let pixel_slice = &mut img.as_slice_mut()[base as usize..base as usize + C];
+                        pixel_slice.copy_from_slice(&fill_color);
+                    }
+                }
+            }
+        }
+    }
+    // drawing the polygon outline
+    draw_polygon(img, points, line_color, thickness);
```

Comment:
i think this is not valuable to have it here; as any user can call themselves first `draw_filled_polygon` and later `draw_polygon` and in this way you are forcing anyone that wants a filled polygon to draw also the outline.

Please, update the function header without the `line_color` and remove this last `draw_polygon` call to have very decoupled functionality.